cms/djangoapps/contentstore/tests/test_outlines.py

```diff
@@ -273,6 +273,40 @@ def test_missing_display_names(self):
         assert outline.sections[0].title == section.url_name
         assert outline.sections[0].sequences[0].title == sequence.url_name
 
+    def test_empty_user_partition_groups(self):
+        """
+        Ignore user partition setting if no groups are associated.
+
+        If we didn't ignore it, we would be greating content that can never be
+        seen by any student.
+        """
+        with self.store.bulk_operations(self.course_key):
+            section = ItemFactory.create(
+                parent_location=self.draft_course.location,
+                category='chapter',
+                display_name='Ch 1',
+                group_access={
+                    49: [],
+                    50: [1, 2],
+                    51: [],
+                }
+            )
+            ItemFactory.create(
+                parent_location=section.location,
+                category='sequential',
+                display_name='Seq 1',
+                group_access={
+                    49: [],
+                }
+            )
+
+        outline, errs = get_outline_from_modulestore(self.course_key)
+        assert len(outline.sections) == 1
+        assert len(outline.sequences) == 1
+        assert outline.sections[0].user_partition_groups == {50: [1, 2]}
+        assert outline.sections[0].sequences[0].user_partition_groups == {}
+        assert len(errs) == 2
```

Comment:
They're collapsed by usage_key. The error message looks like this:

```
<chapter> with url_name="Ch_1" has the following empty group_access user partitions: 49, 51.
This would make this content unavailable to anyone except course staff. Ignoring these
group_access settings when building outline.
```